plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/cache/TestCachingHiveMetastore.java

```diff
@@ -565,6 +567,56 @@ public void testGetTableStatisticsWithoutMetadataCache()
         assertEquals(statsCacheMetastore.getTableStats().getHitRate(), 1.0);
     }
 
+    @Test
+    public void testInvalidateWithLoadInProgress()
+    {
+        CountDownLatch loadInProgress = new CountDownLatch(1);
+        CountDownLatch invalidateDone = new CountDownLatch(1);
+        MockThriftMetastoreClient mockClient = new MockThriftMetastoreClient()
+        {
+            @Override
+            public List<ColumnStatisticsObj> getTableColumnStatistics(String databaseName, String tableName, List<String> columnNames)
+                    throws TException
+            {
+                loadInProgress.countDown();
+                List<ColumnStatisticsObj> result = super.getTableColumnStatistics(databaseName, tableName, columnNames);
+                // get should wait for the invalidation to be done to return result
+                try {
+                    invalidateDone.await(10, SECONDS);
+                }
+                catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+                return result;
+            }
+        };
+        CachingHiveMetastore metastore = createMetastore(mockClient);
+
+        Table table = metastore.getTable(TEST_DATABASE, TEST_TABLE).orElseThrow();
+
+        // invalidate thread
+        new Thread(
+                () -> {
+                    try {
+                        loadInProgress.await(10, SECONDS);
+                    }
+                    catch (InterruptedException e) {
+                        throw new RuntimeException(e);
+                    }
+                    metastore.flushCache();
+                    invalidateDone.countDown();
+                },
+                "invalidate").start();
```

Comment:
i prefer to use executor than starting a thread directly in a test.
then i can wait on the task's Future to ensure the test doesn't leave behind any background tasks, as this could affect other tests.

(the code here simple and benign, but that's a general thinking. Also the code gets copied / evolves)